### PR TITLE
Update docs. Add README.md for new table component.

### PR DESCRIPTION
**Pull request by square-li**
## Overview
Update docs. Add README.md for new table component.
View in another page:
https://github.com/square-li/iot-app-kit/blob/feature-table/docs/Table.md

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: docs/Table.md [hunk 1]
```diff
@@ -17,43 +15,70 @@ const { query } = initialize({ iotsitewiseClient });
 
 // jsx
 <Table
-  viewport={{ duration: '5m' }}
+  viewport = {{ duration: '5m' }}
```
Comment (diehbria) [inline docs/Table.md:18]: remove spacing, not normal syntax for JSX to have spaces around the attribute declarations

File: docs/Table.md [hunk 2]
```diff
@@ -70,88 +95,155 @@ const { query } = initialize({ iotsitewiseClient });
...
 
 ## Properties
 
-The table component contains the following properties that you can customize. 
+The table component contains the following customizable properties:
 
-### `tableColumns`
+Note: Since the table component built over the `@awsui/components-react` table component, please refer to [AWS-UI/table](https://github.com/aws/awsui-documentation/blob/main/components/table.md) for most of the available properties. Differences in properties are listed below.
+
+### `items`
```
Comment (diehbria) [inline docs/Table.md:171]: need more information on `$cellRefs`, what they are, what they are used for, how the data becomes accesible and displayed on the table

File: packages/table/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Table
+
+`@iot-app-kit/table` is a table component built over `@awsui/components-react` components providing client-side sorting, property filtering.
+
+Presents data in a two-dimensional table format, arranged in columns and rows in a rectangular form.
+
+## Stack
```
Comment (diehbria) [inline packages/table/README.md:7]: we have a package.json that declares this. This isn't necessary

**Review by diehbria — CHANGES_REQUESTED** [review]

File: docs/Table.md (hunk 1 above)
Comment (square-li) [reply docs/Table.md]: Removed spaces.

File: docs/Table.md [hunk 4]
```diff
@@ -72,86 +97,183 @@ const { query } = initialize({ iotsitewiseClient });
...
+  wind_speed: {
+    $cellRef: {
+      id: 'dataStream-2',
+      resolution: 0
+    }
+  },
+}
+```
+#### `$cellRef`
+$cellRef is a special keyword in Item. We use it to find according dataStream. For the above example,
```
Comment (boweihan) [inline docs/Table.md:204]: Nit: “the” according dataStream

File: docs/Table.md [hunk 5]
```diff
@@ -72,86 +97,183 @@ const { query } = initialize({ iotsitewiseClient });
...
+      resolution: 0
+    }
+  },
+}
+```
+#### `$cellRef`
+$cellRef is a special keyword in Item. We use it to find according dataStream. For the above example, 
+the item has two properties. Both of them are using `$cellRef` to map properties to data stream.
+
+Other property of an item will remain the same.
```
Comment (boweihan) [inline docs/Table.md:207]: Nit: other “properties”

File: packages/table/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+### Local Setup
+
+```
+yarn install 
+```
+
+### Jest
+
+Jest tests are set up to run with `npm test` or `yarn test`.
```
[the remark this answers is not in the document]
Comment (boweihan) [reply packages/table/README.md:26]: We recommend yarn right above this so let’s just have this say “yarn test”

**Review by boweihan — CHANGES_REQUESTED** [review]

**Review by NorbertNader — APPROVED** [review]

File: docs/Table.md [hunk 7]
```diff
@@ -72,86 +97,183 @@ const { query } = initialize({ iotsitewiseClient });
...
+$cellRef is a special keyword in Item. We use it to find the according dataStream. For the above example, 
+the item has two properties. Both of them are using `$cellRef` to map properties to data stream.
+
+Other properties of an item will remain the same. 
+```typescript jsx
+// If we have defined items like below
+const items: Item[] = [{
+  rpm: {
+    $cellRef: {
+      id: 'dataStream-1',
```
Comment (NorbertNader) [inline docs/Table.md:213]: maybe we could give an example using [toId](https://github.com/awslabs/iot-app-kit/blob/main/packages/source-iotsitewise/src/time-series-data/util/dataStreamId.ts#L6) like [here](https://github.com/awslabs/iot-app-kit/blob/main/docs/Table.md)

**Review by NorbertNader — APPROVED** [review]

**Review by boweihan — APPROVED** [review]
LGTM!

File: docs/Table.md (hunk 2 above)
Comment (square-li) [reply docs/Table.md:173]: added a section for $cellRef

File: docs/Table.md (hunk 7 above)
Comment (square-li) [reply docs/Table.md:221]: Will add that.

File: docs/Table.md [hunk 8]
```diff
@@ -19,41 +19,68 @@ const { query } = initialize({ iotsitewiseClient });
...
+  items={[
     {
-      header: 'Tempature',
-      rows: [
-        toId({ assetId: 'engine-1', propertyId: 'tempature' }),
-        toId({ assetId: 'engine-2', propertyId: 'tempature' }),
-      ]
+      rpm: {
+        $cellRef: {
+          id: toId({ assetId: 'turbine-1', propertyId: 'rotation_per_minute' }),
```
Comment (diehbria) [inline docs/Table.md:26]: we should add the import statement toId to give the full example

File: docs/Table.md [hunk 9]
```diff
@@ -72,86 +99,202 @@ const { query } = initialize({ iotsitewiseClient });
...
-Type: Array
+### `items`
+Specifies the data displayed in rows. Each item contains the data for one row. The display of a row is handled by the cell property of each column definition in the columnDefinitions property.
 
-Each table column contains the following properties
+Type: Array of `Item`
 
-`header`
+#### `Item`
+An item maps its own properties to dataStreams in the following example:
```
Comment (diehbria) [inline docs/Table.md:179]: we don't provide our documentation as typescript defintions, check out how the other docs are written

Comment (diehbria) [reply docs/Table.md:179]: also check out how the typical AWS docs are written, we are trying to adhere to that style the best we can

**Review by diehbria — CHANGES_REQUESTED** [review]

File: docs/Table.md [hunk 10]
```diff
@@ -72,86 +99,215 @@ const { query } = initialize({ iotsitewiseClient });
...
+    }
+  },
+}
+```
+#### `$cellRef`
+$cellRef is a special keyword in Item. We use it to find the according dataStream. For the above example, 
+the item has two properties. Both of them are using `$cellRef` to map properties to data stream.
 
-Type: string
+Type: Object
```
Comment (boweihan) [inline docs/Table.md:210]: $cellRef doesn't have an explicit type?

File: docs/Table.md [hunk 11]
```diff
@@ -72,86 +99,215 @@ const { query } = initialize({ iotsitewiseClient });
...
 
-`rows`
+Required: No
 
-Array of content of each cell within the respective column. Rows are defined in order, the first row item corresponds to the first row on the table that doesn't represent column headers.
+$cellRef have two properties:
```
Comment (boweihan) [inline docs/Table.md:214]: This is how objects are defined in sitewise docs - https://docs.aws.amazon.com/iot-sitewise/latest/APIReference/API_AssetErrorDetails.html - with "Contents" describing the keys.

File: docs/Table.md [hunk 12]
```diff
@@ -72,86 +99,215 @@ const { query } = initialize({ iotsitewiseClient });
...
 
-* A string representing the id of a time series data stream, which you can get by utilizing the time series data source of choices `toId` helper, for example, with AWS IoT SiteWise, you can refer to an id with the following code:
-```
-import { toId } from '@iot-app-kit/source-sitewise';
-const column = {
-  header: 'some-header',
-  rows: [toId({ assetId: 'my-asset', propertyId: 'property-id' })]
-}
-```
+Type: Array of object
```
Comment (boweihan) [inline docs/Table.md:273]: Is there a type for this? also above I believe you had "Object" capitalized vs lowercase here

File: docs/Table.md (hunk 11 above)
Comment (boweihan) [reply docs/Table.md:214]: I see the other docs that we have follow this pattern (i.e. StatusGrid) so maybe it's best to be internally consistent for now.

File: docs/Table.md (hunk 8 above)
Comment (square-li) [reply docs/Table.md:26]: Import statements are the top of the code block.

File: docs/Table.md (hunk 10 above)
Comment (square-li) [reply docs/Table.md:214]: No it doesn't. But we have a type for Item which includes it.

**Pull request merged**